### ISSUE TITLE
hostname not extracted from docker info

### ISSUE THREAD

**Issue by fiws** [DataDog/docker-dd-agent#139]
**Output of the [info page](https://help.datadoghq.com/hc/en-us/articles/203764635-Agent-Status-and-Information) **

```
2016-11-04 16:12:48,773 | WARNING | dd.collector | utils.service_discovery.config(config.py:32) | No configuration backend provided for service discovery. Only auto config templates will be used.
===================
Collector (v 5.9.1)
===================

  Status date: 2016-11-04 16:12:47 (1s ago)
  Pid: 17
  Platform: Linux-3.10.0-327.36.1.el7.x86_64-x86_64-with-debian-8.6
  Python Version: 2.7.12, 64bit
  Logs: <stderr>, /var/log/datadog/collector.log

  Clocks
  ======
  
    NTP offset: Unknown (No response received from 1.datadog.pool.ntp.org.)
    System UTC time: 2016-11-04 16:12:50.125940
  
  Paths
  =====
  
    conf.d: /etc/dd-agent/conf.d
    checks.d: /opt/datadog-agent/agent/checks.d
  
  Hostnames
  =========
  
    socket-hostname: 1f0ad562a8bc
    hostname: 1f0ad562a8bc
    socket-fqdn: 1f0ad562a8bc
  
  Checks
  ======
  
    ntp
    ---
      - Collected 0 metrics, 0 events & 1 service check
  
    disk
    ----
      - instance #0 [OK]
      - Collected 40 metrics, 0 events & 1 service check
  
    redisdb
    -------
      - instance #0 [ERROR]: 'Error connecting to 10.255.0.6:6379. timed out.'
      - instance #1 [ERROR]: 'Error connecting to 172.17.0.13:14379. timed out.'
      - Collected 0 metrics, 0 events & 3 service checks
      - Dependencies:
          - redis: 2.10.3
  
    docker_daemon
    -------------
      - instance #0 [OK]
      - Collected 168 metrics, 1 event & 2 service checks
  
  
  Emitters
  ========
  
    - http_emitter [OK]

===================
Dogstatsd (v 5.9.1)
===================

  Status date: 2016-11-04 16:12:41 (8s ago)
  Pid: 14
  Platform: Linux-3.10.0-327.36.1.el7.x86_64-x86_64-with-debian-8.6
  Python Version: 2.7.12, 64bit
  Logs: <stderr>, /var/log/datadog/dogstatsd.log

  Flush count: 6
  Packet Count: 0
  Packets per second: 0.0
  Metric count: 1
  Event count: 0
  Service check count: 0

===================
Forwarder (v 5.9.1)
===================

  Status date: 2016-11-04 16:12:47 (2s ago)
  Pid: 16
  Platform: Linux-3.10.0-327.36.1.el7.x86_64-x86_64-with-debian-8.6
  Python Version: 2.7.12, 64bit
  Logs: <stderr>, /var/log/datadog/forwarder.log

  Queue Size: 3936 bytes
  Queue Length: 1
  Flush Count: 20
  Transactions received: 7
  Transactions flushed: 6
  Transactions rejected: 0
  
```

**Additional environment details (Operating System, Cloud provider, etc):**

**Steps to reproduce the issue:**
1. start a dd-agent container without the hostname set

**Describe the results you received:**
uses the "docker" hostname (random hex)

**Describe the results you expected:**
use the hostname from `docker info`, as described in the readme.
> By default the agent container will use the Name field found in the docker info command from the host as a hostname

**Additional information you deem important (e.g. issue happens only occasionally):**
I cannot pass -h, because I am using [docker services](https://docs.docker.com/engine/reference/commandline/service_create/) and there currently seems to be no way to pass node specific info.

**Comment by Jwpe** [comment]
I'm also experiencing this issue.

**Comment by anroots** [comment]
I'm running `:kubernetes` on Openshift 3, `5.9` worked fine, upgraded to `5.10` and the agent couldn't identify the underlying hosts' name any more.

![selection_151](https://cloud.githubusercontent.com/assets/469120/21452128/5539f544-c90d-11e6-9312-49d4099e3609.png)

Implemented a quickfix (see [PR#110](https://github.com/DataDog/docker-dd-agent/pull/110/files#diff-bd9b354235ee8d9484a642b6197d8904R4)) by exporting `HOSTNAME` in `/entrypoint.sh`:

```bash
# /entrypoint.sh
export HOSTNAME=`/opt/datadog-agent/embedded/bin/python -c "import docker;print docker.Client(version='auto').info().get('Name', '')"`
```

**Comment by aerostitch** [comment]
Should this be closed with the merge of #110 ?

**Comment by hkaj** [comment]
yep, thanks for the heads-up
